### DISCUSSION TITLE
Add method to update card details for subscriptions.

### DISCUSSION THREAD

**Discussion by grantnhoward** [craftcms/commerce-stripe#162] [Ideas]
Unless I am mistaken, the only way to accomplish this now would be to cancel the subscription and re-subscribe with new card info? If this is not the case please let me know. 

I believe this involves created a valid card token and saving it for the Customer associated with the active Subscription, but would be incredibly useful for providing a way to cleanly update existing subscriptions.
([https://stripe.com/docs/api#update_card](https://stripe.com/docs/api#update_card)) 

Thank you!

**Comment by andris-sevcenko** [comment]
As far as Stripe subscriptions go, the default payment method for the customer is always used to pay for those, so it's only a matter of allowing only one stored card on your frontend to ensure that it's the default for your customer

Currently, there's no way of updating stored payment cards, but you can delete the previous card (which will also delete it on Stripe's end) and add a new payment card (which, being the only one, will become the default card for the user).

I'll keep this issue open because we definitely need to add a way of updating an existing card in one go, though.

**Comment by sjcallender** [comment]
We've created a simple plugin that gets the default payment source for Stripe: [https://github.com/FosterCommerce/commerce-stripe-paymentsource](https://github.com/FosterCommerce/commerce-stripe-paymentsource)

Might be helpful until this is included in the Stripe gateway plugin.
